lib/kasket/read_mixin.rb

```diff
@@ -49,6 +49,14 @@ def find_by_sql_with_kasket_on_id_array(keys)
 
     protected
 
+    def filter_pending_records(records)
+      if pending_records = Kasket.pending_records
+        records.map { |record| pending_records.fetch(record, record) }.compact
```

Comment:
`pending_records[record] || record` wouldn't support the case where `record` is in the map but is `nil`, I actually want the `nil` to be returned in that case